### PR TITLE
feat: Change go-netbox major version

### LINKED ISSUE #143: Update to go-netbox major version v3
## Summary

Update go-netbox to 3.2.2 and change import to major version v3

### PR DESCRIPTION
**Pull request by amhn**
Fixes #143

This PR updates go-netbox to version 3.2.2 and changes the imports to use the new major version.

### CODE REVIEW

File: netbox/data_netbox_json_circuits_circuit_terminations_list.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ func dataNetboxJSONCircuitsCircuitTerminationsListRead(ctx context.Context, d *s
 	params.Offset = &offset
 	for int64(len(tmp)) < desiredLength {
 		offset = int64(len(tmp))
-		if limit > desiredLength-offset {
```
Comment (smutel) [inline netbox/data_netbox_json_circuits_circuit_terminations_list.go:59]: Why this change ?

Comment (amhn) [reply netbox/data_netbox_json_circuits_circuit_terminations_list.go]: Fixed. I did run utils/generateJsonDatasources. That file is now changed to produce the same output as go fmt

**Comment by smutel** [comment]
Thank you @amhn
